modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/DiscountedCumulativeGainAt.java

```diff
@@ -178,19 +193,50 @@ private static double computeDCG(List<Integer> ratings) {
         PARSER.declareInt(DiscountedCumulativeGainAt::setUnknownDocRating, UNKNOWN_DOC_RATING_FIELD);
     }
 
+    @Override
+    public DiscountedCumulativeGainAt fromXContent(XContentParser parser, ParseFieldMatcher matcher) {
+        return DiscountedCumulativeGainAt.fromXContent(parser, new ParseFieldMatcherSupplier() {
+            @Override
+            public ParseFieldMatcher getParseFieldMatcher() {
+                return matcher;
+            }
```

Comment:
> I think this can be simplified to DiscountedCumulativeGainAt.fromXContent(parser, () -> matcher).

Done. Thanks for catching this.

> Also I'm wondering why we need to duplicate the static method. I see this is necessary when implementing 
> FromXContentBuilder, but then again we were able do without this for the query builders for example. If 
> fromXContent() should be a non-static thing, maybe we can get rid of the static implementation instead?

We didn't have that issue in the query builders because those didn't implement FromXContentBuilder. Looking at the type hierarchy of that class I see barely any other class in our codebase actually implement that interface. 

I believe I introduced this interface because initially the roundtrip helper method also contained code that called fromXContent and did the final checks for equality/hashcode etc. In the current state of the tests it's probably not needed anymore. Checking what things look like if I remove implementing that interface now.

What I do wonder is whether this interface should actually be used more widely, or dropped, or else. Ping @imotov @nik9000 who last touched it.